File: MDfiles/Data structures and Algorithms with Java Python and Javascript/JavaScript/testSet.js
function Set(){

    this.dataStore=[];
    this.add = add;
    this.remove=remove;
    this.size=size;
    this.show = show;
    this.contains=contains;
    this.union = union;
    this.intersect= intersect;
    this.difference=difference;
    this.subset = subset;
}

function union (set){
    var tempset = new Set();
    for (var i=0;i<this.dataStore.length;i++){
        tempset.add(this.dataStore[i]);
    }
    for (var j=0;j<set.dataStore.length;j++){
        if (!tempset.contains(set.dataStore[j])){
            tempset.dataStore.push(set.dataStore[j]);
        }
    }
    return tempset;
}

function subset(set){

    if (this.size>set.size){
        return false;
    }else{
        for (var i=0;i<this.dataStore.length;i++){
            if (!set.contains(this.dataStore[i])){
                    return false;
            }
        }
    }
    return true;

}

function difference(set){
    var tempSet = new Set();
    for (var i=0;i<this.dataStore.length;i++){
        if (!set.contains(this.dataStore[i])){
            tempSet.add(this.dataStore[i]);
        }
    }
    return tempSet;
}

function intersect(set){
    var tempset = new Set();
    for (var i=0;i<this.dataStore.length;i++){
        if (set.contains(this.dataStore[i])){
            tempset.add(this.dataStore[i]);
        }
    }
    return tempset;
}

function size (){
    return this.dataStore.length;
}

function contains(item){
    if (this.dataStore.indexOf(item)>-1){
        return true;
    }else{
        return false;
    }
}

function add(item){

    if (this.dataStore.indexOf(item)<0){
        this.dataStore.push(item);
        return true;
    }else{
        return false;
    }
    
}
function remove(item){

    var pos = this.dataStore.indexOf(item);
    if (pos>-1){
        this.dataStore.splice(pos,1);
        return true;
    }else{
        return false;
    }
}

function show(){
    return this.dataStore;
}

var s = new Set();
s.add('A');
s.add('B');
s.add('C');
console.log(s.size())
console.log(s.show());
var dmp = new Set();
dmp.add('A');
dmp.add('E');
dmp.add('F');
console.log(dmp.show());

var it = new Set();
it = s.union(dmp);
console.log('it=>'+it.show());

var inter = s.intersect(dmp);
console.log('inter=>'+inter.show());

var diff = new Set();
diff = s.difference(dmp)
console.log('diff=>'+diff.show());

console.log('subset=>'+s.subset(dmp))

File: MDfiles/Data structures and Algorithms with Java Python and Javascript/Python/testSet.py
import pytest

def test_set():

    print('\n')
    print('test set in python')
    s = Set()
    s.add('A')
    s.add('B')
    s.add('C')
    print(s.show())
    # s.remove('B')
    # print(s.show())

    dim = Set()
    dim.add('A')
    dim.add('B')
    dim.add('C')
    dim.add('D')
    print('dim',dim.show())
    print('dim size',dim.size())

    u = Set()
    u = s.union(dim)
    print('union=>',u.show())

    inter = Set()
    inter = s.intersect(dim)
    print('inter=>',inter.show())

    diff = Set()
    diff = s.difference(dim)
    print('diff=>',diff.show())

    print('subset',s.subset(dim))


class Set:

    def __init__(self):
        self.dataStore=[]

    def add(self,data):
        try:
            if self.dataStore.index(data)>-1:
                return False
        except ValueError:
            self.dataStore.append(data)
            return True

    def remove(self,data):

        pos = self.dataStore.index(data)
        try:
            if pos >-1:
                self.dataStore.remove(self.dataStore[pos])
                return True
        except ValueError:
            return False

    def show(self):
        return self.dataStore

    def size(self):
        return len(self.dataStore)
    
    def contains(self,data):
        try:
            if self.dataStore.index(data)>-1:
                return True
        except ValueError:
            return False

    def union(self,arr):

        tempset = Set()

        for i in range(0,len(self.dataStore)):
            tempset.add(self.dataStore[i])
        for j in range(0,len(arr.dataStore)):
            if tempset.contains(arr.dataStore[j])!=True:
                tempset.dataStore.append(arr.dataStore[j])
        return tempset

    def intersect (self,arr):

        tempset = Set()
        for i in range(0,len(self.dataStore)):
            if arr.contains(self.dataStore[i]):
                tempset.add(self.dataStore[i])
        return tempset

    def difference(self,arr):

        tempset = Set()
        for i in range(0,len(self.dataStore)):
            if arr.contains(self.dataStore[i])!=True:
                tempset.add(self.dataStore[i])
        return tempset

    def subset(self,arr):

        if len(self.dataStore)>len(arr.dataStore):
            return False
        else:
            for i in range(0,len(self.dataStore)):
                if arr.contains(self.dataStore[i])!=True:
                    return False
        return True







        

